docs/docs/icechunk-python/storage.md

```diff
@@ -0,0 +1,222 @@
+    )
+    ```
+
+=== "Anonymous"
+
+    With this option, you connect to S3 anonymously (without credentials).
+    This is suitable for public data. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1,
+        anonymous=True,
+    )
+    ```
+
+=== "Refreshable Credentials"
+
+    With this option, you provide a callback function that will be called to obtain S3 credentials when needed. [See the API](./reference.md#icechunk.s3_storage)
+
+    ```python
+    def get_credentials() -> S3StaticCredentials:
+        return icechunk.S3StaticCredentials(access_key_id="xyz", secret_access_key="abc")
+
+    icechunk.s3_storage(
+        bucket="icechunk-test",
+        prefix="quickstart-demo-1",
+        region='us-east-1',
+        get_credentials=get_credentials,
+    )
+    ```
+
+#### AWS Limitations
+
+- Icechunk is currently incompatible with [S3 Express One Zone](https://aws.amazon.com/s3/storage-classes/express-one-zone/). See [this issue](https://github.com/earth-mover/icechunk/issues/667) for more discussion.
+
+#### Tigris
+
+[Tigris](https://www.tigrisdata.com/) is available as a storage backend for Icechunk. Functionally this storage backend is the same as S3 storage, but with a different endpoint. Icechunk provides a helper function specifically for [creating Tigris storage configurations](./reference.md#icechunk.tigris_storage). There are a few things to be aware of when using Tigris:
+
+- The `endpoint_url` parameter must be set to the Tigris endpoint. This is defaulted to `https://fly.storage.tigris.dev` but may be different depending on your Tigris configuration.
+- Tigris is a globally distributed object store by default. There are currently some bugs with consistency when the store is distributed across multiple regions. For now, it is recommended to use a single region for your Tigris store.
+
+#### Minio
+
+[Minio](https://min.io/) is available as a storage backend for Icechunk. Functionally this storage backend is the same as S3 storage, but with a different endpoint.
+
+For example, if we have a Minio server running at `http://localhost:9000` with access key `minio` and
+secret key `minio123` we can create a storage configuration as follows:
+
+```python
+icechunk.s3_storage(
+    bucket="icechunk-test",
+    prefix="quickstart-demo-1",
+    region='us-east-1',
+    access_key_id='minio',
+    secret_access_key='minio123',
+    endpoint_url='http://localhost:9000',
+    allow_http=True,
+```
+
+A few things to note:
+
+1. The `endpoint_url` parameter is set to the URL of the Minio server.
+2. If the Minio server is running over HTTP and not HTTPS, the `allow_http` parameter must be set to `True`.
+3. Even though this is running on a local server, the `region` parameter must still be set to a valid region. By default Minio uses `us-east-1`.
```

Comment:
I don't know about the default, we should try, but probably you can pass anything for the single node MinIO.

Reply:
adding a link to this: https://github.com/minio/minio/discussions/15063